pa

```diff
@@ -251,12 +251,12 @@ main() {
     # state on exit or Ctrl+C.
     [ -t 1 ] && trap 'stty echo icanon' INT EXIT
 
-    case $1 in
-    a*) pw_add "$2" ;;
-    d*) pw_del "$2" ;;
-    e*) pw_edit "$2" ;;
+    case $command in
+    a*) pw_add ;;
```

Comment:
There are actually no local variables in a shell as defined by POSIX, see https://pubs.opengroup.org/onlinepubs/9699919799/xrat/V4_xcu_chap02.html#tag_23_02_09_18

> Local variables within a function were considered and included in another early proposal (controlled by the special built-in **_local_**), but were removed because they do not fit the simple model developed for functions and because there was some opposition to adding yet another new special built-in that was not part of historical practice

So technically it doesn't matter where `name` is assigned, it would still be defined globally. It's not required that functions use `$name` from `main`, I can remove its assignment right after `shift`, pass `"$*"` to functions here, and leave `name=$1` in `pw_add` and `pw_edit` as is. I'm not sure it would be better for readability though, as the way I've done it glob guards and all of the functions get named variables for free, which IMO reads better than using positional arguments

Reply:
ok, make sense.
thx for clarifications regarding local vars in the posix definition.